### DISCUSSION TITLE
Is there a step by step tutorial about to build an extension like panel, layout, modules, etc..?

### DISCUSSION THREAD

**Discussion by biapar** [directus/directus#11023] [Archive]
Is there a step by step tutorial about to build an extension like panel, layout, modules, etc..?

**Comment by oreilles** [comment]
Tutorials and code samples for each type of extension are available in the corresponding section of the docs: https://docs.directus.io/extensions/introduction/

**Reply by biapar** [reply]
I followed that link and I'm able to create a simple module with only a title. I wish create a module that filter a collection and export data in Excel. I don't find any documentation about to use the default DIrectus date picker and tutorials/recipes how to extent Directus. There a lots on Strapi.

**Reply by oreilles** [reply]
We currently have people on the team working full time on improving the docs, please tell us if there's some specific examples you'd like to see and we'll try to follow up. 👍

**Reply by biapar** [reply]
An example is to create a module and get a collection data filtered by some inputs ( data from , to ) and then to export in some format ( Excel ? ). 
I 'm trying to do this. I've only the problem that I don't know how to use v-date-picker and I don't want to use an external date picker component.  I get collection data from API.

**Reply by oreilles** [reply]
Directus itself is relying on [Flatpickr](`https://flatpickr.js.org/`), so you might want to use it. If you specifically want to use Directus `v-date-picker`, you should be able to just import that component from your module vue file with `import VDatePicker from 'relative/path/to/v-date-picker';`

**Reply by biapar** [reply]
I see the calendar but there isn't the field.

![image](https://user-images.githubusercontent.com/661517/149354677-184af61c-418e-4ff6-aa1e-42a8a31ef673.png)

![image](https://user-images.githubusercontent.com/661517/149354808-7bccb4b0-c340-4f26-88eb-131fc8f07dfd.png)

**Reply by oreilles** [reply]
If you want to import the full interface, it is located [here](https://github.com/directus/directus/blob/main/app/src/interfaces/datetime/datetime.vue).

**Reply by biapar** [reply]
> If you want to import the full interface, it is located [here](https://github.com/directus/directus/blob/main/app/src/interfaces/datetime/datetime.vue).

I think that it's a bit complex for a simple date input field.

**Reply by oreilles** [reply]
But you just have to import the file. If you think the implementation is too complex, feel free to write one that fits your requirements 👍

**Reply by biapar** [reply]
How to call after import? I don't understand. Thanks.
I get data -:), now. I've only to implement the data filter and Excel export.

**Reply by oreilles** [reply]
In order to convert JSON data to XLS, you could use [this package](https://www.npmjs.com/package/xlsx).
If you need help on how to import and use Vue components, there are very good tutorials on the official vue website and elsewhere on the internet.
This help forum is intended for problems that are directly related to Directus - If you need help regarding usage of a specific library of framework, please use the documentation and help forum that target those.
